sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/sources/RateStreamMicroBatchStream.scala

```diff
@@ -67,6 +67,7 @@ class RateStreamMicroBatchStream(
           writer.write("v" + VERSION + "\n")
           writer.write(metadata.json)
           writer.flush
+          writer.close
```

Comment:
I think that will close the underlying output stream, and I don't think that's what this is supposed to do here.